compiler-rt/lib/asan/CMakeLists.txt

```diff
@@ -136,7 +160,7 @@ append_list_if(MINGW "${MINGW_LIBRARIES}" ASAN_DYNAMIC_LIBS)
 add_compiler_rt_object_libraries(RTAsan_dynamic
   OS ${SANITIZER_COMMON_SUPPORTED_OS}
   ARCHS ${ASAN_SUPPORTED_ARCH}
-  SOURCES ${ASAN_SOURCES} ${ASAN_CXX_SOURCES}
+  SOURCES ${ASAN_SOURCES}
```

Comment:
this line seems to be the problem, reverting it fixes the tests

Reply:
This should be safe to revert, and I'll make a PR reverting it (and probably just commit it if it doesn't get any reviews before I go to sleep).

Reply:
I've opened https://github.com/llvm/llvm-project/pull/108329 which should revert this particular edit (which was arguably a driveby change anyway)